### DISCUSSION TITLE
Default settings for search are extremely slow

### DISCUSSION THREAD

**Discussion by Natenom** [razonyang/hugo-theme-bootstrap#516] [Ideas]
The default settings for FuseJS in the theme result in an extremly slow and cpu intense search.

For my blog (~2600 Posts) it took often so long that the browser asked to close tab (because a script caused to much load, 100% CPU). And the results were wrong most often. There where blog posts (hundrets sometimes) in the result list that didn't have the searched word included.

Today I changed the default settings to (and disabled fuzzy search though):

```
search.fuse.threshold = 0.0
search.fuse.location = 0
search.fuse.distance = 100
search.fuse.minMatchCharLength = 3
```

Now the search results appears within 1 second in Chromium and a bit slower (few seconds) in Firefox.

I also enabled extended search for FuseJS.
```
search.fuse.useExtendedSearch = true
```

The default settings are probably sufficient for smaller blogs, but at some point the default settings should be changed.

Could you add an information about this to your documentation, please?

You can try the new settings on my [blog](https://blog.natenom.com/search/?q=hugo%20bootstrap).

Thank you :)

**Comment by chrisrummel** [comment]
Those are exactly the same settings I came to use after I switched to hugo-theme-bootstrap, although I never noticed any performance issues.
The search results are still completey unusable, though. I don't or only rarely get hits for the word I searched for, but for any permutation of the letters the word is made of. You can imagine the mess of irrelevant results... :-( 
But that is a topic that doesn't belong here...

**Comment by razonyang** [comment]
I changed the `minMatchCharLength` as `3`, `threshold` as default(`0.6`). The [docs](https://hbs.razonyang.com/en/docs/layouts/search/) include the link of  the [Fuse.js Options](https://fusejs.io/api/options.html) already.
